Add Azure Databricks examples

hwang-db opened the pull request:
This is the first PR to migrate additional examples into this repo, including:
1. adb-exfiltration-protection
2. adb-external-hive-metastore
3. adb-private-links
4. adb-with-squid-proxy
5. adb-splunk
6. adb-git-proxy
7. adb-uc (with automation on account admin + spn set up etc)
With documentation in each subfolder.

alexott reviewed (changes requested):
We need to remove `terraform.tfstate.*` files from PR.

Also, can we reorganize these into modules + calls to modules, then it would be easier for people to reuse the code?

hwang-db commented:
Thanks @alexott I've updated .gitignore and removed states files. For the modules I will work on them to make some reusable modules in follow up PRs; there are also some examples that might not need a module and remain as an individual example itself.
